src/random.cpp

```diff
@@ -389,6 +391,8 @@ class RNGState {
             ++m_counter;
             // Finalize the hasher
             hasher.Finalize(buf);
+            // Ensure Finalize() populated buf
+            assert(buf);
```

Comment:
FWIW: asserting the *address* of the buffer, on the stack, like you do here does nothing.

Reply:
Oops. Looping through it doesn't sound appealing either.

Reply:
Yikes.

Reply:
For future reference, something like this works (?), though it's hideous, probably needs a macro, which adds even more complexity:
```
unsigned char zeros[sizeof(buf)] = {0};
assert(!memcmp(fbuf, zeros, sizeof(buf)));
```

Reply:
Funny who you run into while googling that: https://rusty.ozlabs.org/?p=560 :-)